src/service/quoting-engine.ts

```diff
@@ -94,37 +94,52 @@ export class QuotingEngine {
             return null;
         }
         var targetBasePosition = tbp.data;
-        
+
         var latestPosition = this._positionBroker.latestReport;
         var totalBasePosition = latestPosition.baseAmount + latestPosition.baseHeldAmount;
-        
+
         if (totalBasePosition < targetBasePosition - params.positionDivergence) {
             unrounded.askPx = null;
             unrounded.askSz = null;
             if (params.aggressivePositionRebalancing)
-                unrounded.bidSz = Math.min(params.aprMultiplier*params.size, targetBasePosition - totalBasePosition);
+                unrounded.bidSz = Math.min(params.aprMultiplier*params.buySize, targetBasePosition - totalBasePosition);
         }
-        
+
         if (totalBasePosition > targetBasePosition + params.positionDivergence) {
             unrounded.bidPx = null;
             unrounded.bidSz = null;
             if (params.aggressivePositionRebalancing)
-                unrounded.askSz = Math.min(params.aprMultiplier*params.size, totalBasePosition - targetBasePosition);
+                unrounded.askSz = Math.min(params.aprMultiplier*params.sellSize, totalBasePosition - targetBasePosition);
         }
-        
+
         var safety = this._safeties.latest;
         if (safety === null) {
             this._log.warn("cannot compute a quote since trade safety is not yet computed!");
             return null;
         }
-        
-        if (params.mode === Models.QuotingMode.PingPong) {
+
+        if (params.mode === Models.QuotingMode.PingPong || params.mode === Models.QuotingMode.Boomerang) {
           if (unrounded.askSz && safety.buyPing && unrounded.askPx < safety.buyPing + params.width)
             unrounded.askPx = safety.buyPing + params.width;
           if (unrounded.bidSz && safety.sellPong && unrounded.bidPx > safety.sellPong - params.width)
             unrounded.bidPx = safety.sellPong - params.width;
         }
-        
+
+        if (unrounded.askPx !== null)
+          for (var fai = 0; fai < filteredMkt.asks.length; fai++) {
+            if (filteredMkt.asks[fai].price > unrounded.askPx) {
+              unrounded.askPx = filteredMkt.asks[fai].price - .01;
+              break;
+            }
+          }
+        if (unrounded.bidPx !== null)
+          for (var fbi = 0; fbi < filteredMkt.bids.length; fbi++) {
+            if (filteredMkt.bids[fbi].price < unrounded.bidPx) {
+              unrounded.bidPx = filteredMkt.bids[fbi].price + .01;
+              break;
+            }
+          }
+
```

Comment:
price accomodation to avoid "hollows" in the book and attempt to get "as much as possible"